### ISSUE TITLE
Fix S4144 FN: Should raise for methods in interfaces and structs

### ISSUE THREAD

**Issue by mary-georgiou-sonarsource** [SonarSource/sonar-dotnet#5128]
### Description

Rule S4144 does not raise an issue for [methods implemented in interfaces](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#default-interface-methods) and for methods implemented in structs.

### Repro steps

```csharp
interface SomeInterface
{
    void Foo1() // FN
    {
        string s = "test";
        Console.WriteLine("Result: {0}", s);
    }

    void Foo2() // FN
    {
        string s = "test";
        Console.WriteLine("Result: {0}", s);
    }
}

struct SomeStruct
{
    void Bar1() // FN
    {
        string s = "test";
        Console.WriteLine("Result: {0}", s);
    }

    void Bar2() // FN
    {
         string s = "test";
         Console.WriteLine("Result: {0}", s);
    }
}
```

### Expected behavior

For the given example there should be an issue raised for `Foo1()` and `Foo2()` as they have identical implementation.
Same for the methods `Bar1()` and `Bar2()` .

### Actual behavior

No issue was raised.

### Related information

C#/VB.NET Plugins version: [8.32](https://github.com/SonarSource/sonar-dotnet/releases/tag/8.32.0.39516)
